spec/features/post_auth_spec.rb

```diff
@@ -102,6 +102,21 @@
           expect(last_response.status).to eq(404)
         end
       end
+
+      context 'Given parameters are missing from the GET request' do
+        let(:authentication_result) { 'Access-Accept' }
+        let(:username) { '' }
+        let(:called_station_id) { '' }
+        let(:site_ip_address) { '' }
+
+        it 'returns a 204 status code' do
+          expect(last_response.status).to eq(204)
+        end
+
+        it 'creates a session record' do
+          expect(Session.all.count).to eq(1)
+        end
+      end
```

Comment:
It implicitly tests the `if` condition added in this commit, right? It would throw 500 if we were to delete that `if` statement? 🤔

Reply:
Yes it would throw a 500 with a no_method error on nil.
Having said that, I have never seen the username not be present in one of these requests, but this mirrors how the current implementation deals with this.

Reply:
Think a 400 would be more appropraite in the future, but we can go ahead with this for now.

Reply:
It does still create a record.  See production database, sessions table, id: 139703209.